src/NuGet.Core/NuGet.Commands/Utility/CommandRunnerUtility.cs

```diff
@@ -67,7 +67,15 @@ public static string GetApiKey(ISettings settings, string endpoint, string sourc
             return apiKey ?? defaultApiKey;
         }
 
-        public static async Task<PackageUpdateResource> GetPackageUpdateResource(IPackageSourceProvider sourceProvider, string source)
+        public static async Task<PackageUpdateResource> GetPackageUpdateResource(IPackageSourceProvider sourceProvider, PackageSource packageSource)
```

Comment:
Why is this change made? Why need to separate it?

Reply:
If you look at the runner code, I use the effective PackageSource to check for http. 
I don't want to create the PackageSource twice.